### ISSUE TITLE
Getting error when using FFbase.FFmpegDir on mac

### ISSUE THREAD

**Issue by devna13** [tomaszzmuda/Xabe.FFmpeg#44]
I'm including `http://ffmpeg.zeranoe.com/builds/`  in my apps bin folder and set the `Xabe.FFmpeg.FFbase.FFmpegDir` to the right folder (mac64/win64) based on the OS.

![bin2](https://user-images.githubusercontent.com/4080343/33688656-691c5482-da99-11e7-8307-63e462646018.png)

It works fine on windows 10 but on mac I get this exception:

```
Unhandled Exception: System.ComponentModel.Win32Exception: Permission denied
   at Interop.Sys.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Int32& lpChildPid, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at Xabe.FFmpeg.FFbase.RunProcess(String args, String processPath, Boolean rStandardInput, Boolean rStandardOutput, Boolean rStandardError)
   at Xabe.FFmpeg.FFprobe.RunProcess(String args)
   at Xabe.FFmpeg.FFprobe.GetStream(String videoPath)
   at Xabe.FFmpeg.FFprobe.GetProperties(String videoPath)
   at Xabe.FFmpeg.MediaInfo.get_Properties()
   at Xabe.FFmpeg.MediaInfo.ToString()
   at .Program.<RunApp>d__3.MoveNext() in /Users/<path...>/Program.cs:line 100
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

any thoughts?

Also, I'm new to ffmpeg, do I need to include `ffserver` and `ffplay`  in that directory?

Thanks

**Comment by tomaszzmuda** [comment]
Hello.
ffserver and ffplay are not used by Xabe.FFmpeg. You don't have to include it.
Could you tell me what version of library do you use? Are you using source code directly?

**Comment by devna13** [comment]
I'm using in netcoreapp2.0 console app.

<PackageReference Include="Xabe.FFmpeg" Version="2.2.9" />

**Comment by tomaszzmuda** [comment]
Sorry, I wasn't clear. What version of Xabe.FFmpeg do you installed trough nuget.

**Comment by devna13** [comment]
I'm using 2.2.9 lib in netcoreapp2.0 console app.

`<PackageReference Include="Xabe.FFmpeg" Version="2.2.9" />`

**Comment by tomaszzmuda** [comment]
I think this is problem with permissions. Probably dotnet core doesn't have permissions for that folder.
Could you run this program as sudo?

**Comment by devna13** [comment]
I'm getting same permission exception with `sudo dotnet run`...

**Comment by tomaszzmuda** [comment]
Please post line where you assign path to Xabe.FFmpeg.FFbase.FFmpegDir

**Comment by devna13** [comment]
@tomaszzmuda 

```
//TODO
bool isMac = System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
Xabe.FFmpeg.FFbase.FFmpegDir = Path.GetDirectoryName(Assembly.GetEntryAssembly().Location) + "/ffmpeg/" + (isMac ? "mac64/" : "win64/");
var exist = Directory.Exists(Xabe.FFmpeg.FFbase.FFmpegDir);
Console.WriteLine(exist + "-" + Xabe.FFmpeg.FFbase.FFmpegDir);
```
` Output:  True-/Users/...../bin/Debug/netcoreapp2.0/ffmpeg/mac64/`



later in the code this line throws an exception (not on windows)


```
bool result = await ConversionHelper.ExtractAudio(taskInfo.VideoFullPath, taskInfo.Mp3FullPath).Start();
//taskInfo.VideoFullPath = "/Users/..../bin/Debug/netcoreapp2.0/vid.mp4"
```

```
Exception has occurred: CLR/System.ComponentModel.Win32Exception
An exception of type 'System.ComponentModel.Win32Exception' occurred in System.Private.CoreLib.dll but was not handled in user code: 'Permission denied'
   at Interop.Sys.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Int32& lpChildPid, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at Xabe.FFmpeg.FFbase.RunProcess(String args, String processPath, Boolean rStandardInput, Boolean rStandardOutput, Boolean rStandardError)
   at Xabe.FFmpeg.FFmpeg.<>c__DisplayClass9_0.<RunProcess>b__0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
```

**Comment by tomaszzmuda** [comment]
Try use Path.Combine instead of 
`Xabe.FFmpeg.FFbase.FFmpegDir = Path.GetDirectoryName(Assembly.GetEntryAssembly().Location) + "/ffmpeg/" + (isMac ? "mac64/" : "win64/");`
System.ComponentModel.Win32Exception occurs when .NET cannot start process. I tried to reproduce this bug but in Linux (I don't have access to mac) everythig works. If .NET doesn't have permissions to FFmpeg executables it throw UnauthorizedAccessException now System.ComponentModel.Win32Exception so if it works on Windows but not mac it looks like problem with path. 
Use this:
`Xabe.FFmpeg.FFbase.FFmpegDir = Path.Combine(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location), "ffmpeg", (isMac ? "mac64/" : "win64/"));``

**Comment by devna13** [comment]
Hi @tomaszzmuda  
I tried your suggestion but I still getting same error on mac. 

So I created a simple console app, this time referencing directory ffmpeg outside the bin folder,  it runs fine on windows 10 and it failed on mac with same error.
[https://github.com/devna13/XabeFFmpegMacTest](url)


**But I found the solution, the ffmpeg folder needs elevated permissions** 
**chmod -R 777 'ffmpegdirectorypath'**

```
mac64 $./ffmpeg
-bash: ./ffmpeg: Permission denied
mac64 $chmod -R 777 /Users/code/XabeFFmpegMacTest/XabeFFmpegMacTest/ffmpeg/
mac64 $./ffmpeg
ffmpeg version 3.4 Copyright (c) 2000-2017 the FFmpeg developers
  built with Apple LLVM version 9.0.0 (clang-900.0.37)
```

after that change it started to work!

Thanks @tomaszzmuda !
